assets/src/components/Sidebar/CampaignSidebar.js

```diff
@@ -86,6 +86,30 @@ export class CampaignSidebar extends Component {
       meta: null,
       parent: null,
     };
+    this.handleThemeSwitch = this.handleThemeSwitch.bind( this );
+    this.loadTheme = this.loadTheme.bind( this );
+  }
+
+  async handleThemeSwitch( metaKey, value, meta ) {
+    await this.loadTheme( value )
+    const invalidatedFields = this.state.theme.fields.filter( field => {
+      const resolvedField = getFieldFromTheme(this.state.theme, field.id, meta);
+
+      if ( !resolvedField || !resolvedField.options ) {
+        return !!meta[ field.id ];
+      }
+
+      return !(resolvedField.options.some( option => option.value === meta[ field.id ]) );
```

Comment:
This logic seems to be repeated below, is there a way to encapsulate it better, or make sure it only happens in one place? There also a similar check in `withPostMeta` (not shown in this diff).